[docs] Updating guide to using custom fonts (#7130)
expo/expo: docs/pages/versions/unversioned/guides/using-custom-fonts.md

title: Using Custom Fonts
---

import SnackEmbed from '~/components/plugins/SnackEmbed';

Both iOS and Android and most desktop operating systems come with their own set of platform fonts but if you want to inject some more brand personality into your app, a well picked font can go a long way. And since each
operating system has its own set of platform fonts, if you want to produce an experience that is consistent for all users, you'll want to use your own fonts in your project. This guide will show you how to do that.

## A Minimal but Complete Working Example

> <SnackEmbed snackId="@ccheever/custom-font-example" />

Reviewer: ideally we want to be able to embed snacks inline because they are versioned properly and anyone can send a pr to fix.  see an example of this here: https://github.com/expo/expo/blame/master/docs/pages/versions/v36.0.0/tutorial/text.md#L14

that said, it's not currently possible to include assets in an inline snack. it may be a good project for you if you're interested @ccheever to add the capability to specify an asset to download and include in the project in a query param

Author: Will take a look at this in the future; seems like a good idea and I def prefer inline snacks, esp. for docs.

> 💡 **This won't preview properly in the web viewer of Snack but works everywhere else.** This is just a bug in Snack that should be fixed soon. This code will work on web in expo-cli apps and on all other platforms in Snack.

If you load this up on a device, you should see something that looks like this:

<img src="/static/images/font-example-custom-font.png" style={{maxWidth: 305}} />

Inter Black is very bold and dark and pretty distinctive so you should be able to tell if you're able to 

> get the example working right, or if something is wrong. If the Platform Default font looks a little different 

Reviewer: ```suggestion
get the example working right, or if something is wrong. If the platform default font looks a little different 
```

> for you, that's fine; the platform default font varies from platform to platform.

Reviewer: ```suggestion
for you, that's fine; the platform default font can vary depending on the operating system and the device manufacturer (on Android).
```

## Getting a Font

The first thing you'll need is a font file. In this example, we use Inter Black from the free and open source [Inter font family](https://rsms.me/inter/) by Rasmus Anderson.

You'll want either a font in either `.otf` or `.ttf` format. Those are the two formats that work across all Expo platforms (web, iOS, and Android). If you have both an OTF and a TTF file, just use OTF. It's a newer format that is often smaller and sometimes higher quality in some contexts. Either one is fine though.

The convention in Expo apps is to put your fonts in an `./assets/fonts` directory, but you can put them anywhere you like.

> In this example (which you can see if you [open the Snack in a dedicated window](https://snack.expo.io/@ccheever/custom-font-example)), we've put our font file, `Inter-Black.otf` in the `./assets/fonts` subdirectory of the project.

Reviewer: same thing re: embedding code in the markdown

## Using the `useFonts` Hook

> The example in this guide uses the `useFonts` hook from the [`use-expo`](https://github.com/byCedric/use-expo) hooks library by Cedic van Putten. It is the easiest way to load custom fonts in modern React.

Reviewer 2: ```suggestion
The example in this guide uses the `useFonts` hook from the [`use-expo`](https://github.com/byCedric/use-expo) hooks library by Cedric van Putten. It is the easiest way to load custom fonts in modern React.
```
